enhancements/hypershift/node-lifecycle.md

```diff
@@ -0,0 +1,198 @@
+- As a Service Provider I want to have the ability upgrade the OS and Kube version of the Nodes by recreating the instances, so Hosts are recycled regularly for security hygiene.
+- As a Service Provider I want to have the ability to declaratively specify different pools of Nodes, so I can setup topologies tolerant to domain failures.
+- As a Service Provider I want to have the ability to declaratively specify different pools of Nodes, so I can allocate workloads with concrete architecture or hardware demands.
+- As a Service Provider I want to have the ability to upgrade the control plane independently of the data plane, so they can satisfy different compliance policies.
+- As a cluster Admin I want to have the ability to manipulate Nodes with Kubernetes high level primitives e.g. taints, labels, etc. Without interacting the underlying infra.
+
+### Goals
+- Provide a consumable API for Node management that satisfies HyperShift form factor (multi-tenancy and control plane/data plane decoupling) supporting the following features:
+  - Scaling.
+  - Auto scaling.
+  - Auto repair.
+  - Auto approval.
+  - OS upgrades.
+  - Config updates.
+  - Management/workload cluster network and infra separation.
+  - Multi-tenancy.
+
+### Non-Goals
+
+## Proposal
+### API
+[NodePool](https://github.com/openshift/hypershift/blob/27c0a432bdc8d702f1cdb2a2f3f25e5ae6fbee7d/api/v1alpha1/nodepool_types.go) is the consumer facing API exposed for Node management.
+Its reason to exist is to preserve our ability to satisfy consumer needs and evolve at our own pace while reusing and relying on battle proven technologies for the implementation details.
+
+### Automated compute capacity management
+#### Cluster API
+Having a NodePool API creates the need for machinery to satisfy their intent. We choose to leverage [CAPI](https://github.com/kubernetes-sigs/cluster-api) as the NodePool implementation.
+It's a project that we have been involved since early stages and contains the learnings from running the `machine.openshift.io` in production since OCP 4 release.
+CAPI is now v1beta1 which gives us the API guarantees we require for customer facing APIs.
+It solves the main problems intrinsic to the HyperShift form factor paradigm in this area: 
+- Cloud agnostic Automated machine management.
+- Awareness of management vs workload cluster.
+- Ability to signal Externally managed infrastructure.
+- Pluggable control plane implementation.
+- Pluggable host bootstrapping implementation.
+
+#### Implementation
+The NodePool controller is meant to be a thin layer that proxies API input into CAPI resources and delegates as much as possible of any business logic implementation into CAPI controllers.
+When a HostedCluster is created a CAPI Cluster and a HostedControlPlane are created as a representation in the control plane namespace. This Cluster CR satisfies the contract for any CAPI Machine scalable resource.
+
+When a new NodePool is created:
```

Comment:
We have the workflow section below where the guidelines ask to be explicit about the actor in each step. It's helpful to know who is excepted to do what (human with a specific persona, system component, etc.). You don't have to move all of this content, but could you add some of that sort of detail?

Reply:
Updated.